// Generated from scheduler_command_envelope.ifex by covesa-ifex-core
// DO NOT EDIT - regenerate with: ./generate_proto.sh
syntax = "proto3";



package swdv.scheduler_command_envelope;

// Namespace: scheduler_command
// Scheduler command types for cloud-to-vehicle communication

// Types of scheduler commands
enum command_type_t {
  COMMAND_UNKNOWN = 0;
  COMMAND_CREATE_JOB = 1;
  COMMAND_UPDATE_JOB = 2;
  COMMAND_DELETE_JOB = 3;
  COMMAND_PAUSE_JOB = 4;
  COMMAND_RESUME_JOB = 5;
  COMMAND_TRIGGER_JOB = 6;
}

// Job definition for create operations
message job_definition_t {
  string job_id = 1;
  string title = 2;
  string service = 3;
  string method = 4;
  string parameters_json = 5;
  string scheduled_time = 6;
  string recurrence_rule = 7;
  string end_time = 8;
  string service_address = 9;
}

// Partial job update - only specified fields are updated
message job_update_t {
  string job_id = 1;
  string title = 2;
  string scheduled_time = 3;
  string recurrence_rule = 4;
  string parameters_json = 5;
  string end_time = 6;
}

// Scheduler command envelope (sent from cloud to vehicle)
message scheduler_command_t {
  string command_id = 1;
  uint64 timestamp_ns = 2;
  string requester_id = 3;
  command_type_t type = 4;
  job_definition_t create_job = 5;
  job_update_t update_job = 6;
  string delete_job_id = 7;
  string pause_job_id = 8;
  string resume_job_id = 9;
  string trigger_job_id = 10;
}

// Command acknowledgment (sent from vehicle to cloud)
message scheduler_command_ack_t {
  string command_id = 1;
  bool success = 2;
  string error_message = 3;
  string job_id = 4;
  uint64 timestamp_ns = 5;
}


